patches/minecraft/net/minecraft/server/level/ServerPlayer.java.patch

```diff
@@ -150,6 +150,14 @@
     }
  
     public boolean m_143432_() {
+@@ -1265,6 +_,7 @@
+    public void m_9213_(Entity p_9214_) {
+       Entity entity = this.m_8954_();
+       this.f_8926_ = (Entity)(p_9214_ == null ? this : p_9214_);
++      if (this.f_8926_ instanceof net.minecraftforge.entity.PartEntity<?> partEntity) this.f_8926_ = partEntity.getParent(); // FORGE: fix MC-46486
```

Comment:
So while this could be fine, is there any reason why the parent of the part entity is not another part entity? Or is this just an assumed thing with the PartEntity API that the parent will always be the actual entity?

Reply:
I feel like part entities can't actually have their own part entities. From my experience they're really buggy and can't even act like a normal entity. I should probably still test it out though.

Reply:
Well give it a test today so it can get in for the 1.17 RB. I don't see any reason why it couldn't be true, but if you determine otherwise, just give a reason for it.

Reply:
I'll test it out tomorrow after school since I gtg now.

Reply:
@ChampionAsh5357 
I have an idea. Maybe I can check for the entity. If it's an instance of parent entity, I'll rerun the setCamera method with the parent and if it isn't I'll continue executing it.

Reply:
That's not a good idea. You could just as simply make it a while loop that just says while the entity is a part, set the entity to the parent.

Reply:
Yea that's what I thought and that's why I commented before actually doing it.
Anyways, I changed it.